core/src/main/scala/com/nvidia/spark/rapids/tool/Platform.scala

```diff
@@ -67,73 +82,120 @@ case class InstanceInfo(cores: Int, memoryMB: Long, name: String, numGpus: Int)
+      InstanceInfo(16, 110 * 1024, "Standard_NC16as_T4_v3", 1, T4Gpu), // 1 GPU
+    NodeInstanceMapKey("Standard_NC64as_T4_v3") ->
+      InstanceInfo(64, 440 * 1024, "Standard_NC64as_T4_v3", 4, T4Gpu)  // 4 GPUs
   )
 
   // dataproc and dataproc-gke
   // Google supports 1, 2, or 4 Gpus of most types
   // added to n1-standard boxes. You may be able to add 8 v100's but we
   // are going to ignore that.
-  val DATAPROC_BY_GPUS_CORES = Map(
-    (1, 1) -> InstanceInfo(1, 1 * 3840, "n1-standard-1", 1),
-    (1, 2) -> InstanceInfo(2, 2 * 3840, "n1-standard-2", 1),
-    (1, 4) -> InstanceInfo(4, 4 * 3840, "n1-standard-4", 1),
-    (1, 8) -> InstanceInfo(8, 8 * 3840, "n1-standard-8", 1),
-    (1, 16) -> InstanceInfo(16, 16 * 3840, "n1-standard-16", 1),
-    (1, 32) -> InstanceInfo(32, 32 * 3840, "n1-standard-32", 1),
-    (1, 64) -> InstanceInfo(64, 64 * 3840, "n1-standard-64", 1),
-    (1, 96) -> InstanceInfo(96, 96 * 3840, "n1-standard-96", 1),
-    (2, 1) -> InstanceInfo(1, 1 * 3840, "n1-standard-1", 2),
-    (2, 2) -> InstanceInfo(2, 2 * 3840, "n1-standard-2", 2),
-    (2, 4) -> InstanceInfo(4, 4 * 3840, "n1-standard-4", 2),
-    (2, 8) -> InstanceInfo(8, 8 * 3840, "n1-standard-8", 2),
-    (2, 16) -> InstanceInfo(16, 16 * 3840, "n1-standard-16", 2),
-    (2, 32) -> InstanceInfo(32, 32 * 3840, "n1-standard-32", 2),
-    (2, 64) -> InstanceInfo(64, 64 * 3840, "n1-standard-64", 2),
-    (2, 96) -> InstanceInfo(96, 96 * 3840, "n1-standard-96", 2),
-    (4, 1) -> InstanceInfo(1, 1 * 3840, "n1-standard-1", 4),
-    (4, 2) -> InstanceInfo(2, 2 * 3840, "n1-standard-2", 4),
-    (4, 4) -> InstanceInfo(4, 4 * 3840, "n1-standard-4", 4),
-    (4, 8) -> InstanceInfo(8, 8 * 3840, "n1-standard-8", 4),
-    (4, 16) -> InstanceInfo(16, 16 * 3840, "n1-standard-16", 4),
-    (4, 32) -> InstanceInfo(32, 32 * 3840, "n1-standard-32", 4),
-    (4, 64) -> InstanceInfo(64, 64 * 3840, "n1-standard-64", 4),
-    (4, 96) -> InstanceInfo(96, 96 * 3840, "n1-standard-96", 4)
+  val DATAPROC_BY_INSTANCE_NAME: Map[NodeInstanceMapKey, InstanceInfo] = Map(
+    // g2-standard instances with Intel GPUs
+    NodeInstanceMapKey("g2-standard-4") ->
+      InstanceInfo(4, 16 * 1024, "g2-standard-4", 1, L4Gpu),
+    NodeInstanceMapKey("g2-standard-8") ->
+      InstanceInfo(8, 32 * 1024, "g2-standard-8", 1, L4Gpu),
+    NodeInstanceMapKey("g2-standard-12") ->
+      InstanceInfo(12, 48 * 1024, "g2-standard-12", 1, L4Gpu),
+    NodeInstanceMapKey("g2-standard-16") ->
+      InstanceInfo(16, 64 * 1024, "g2-standard-16", 1, L4Gpu),
+    NodeInstanceMapKey("g2-standard-24") ->
+      InstanceInfo(24, 96 * 1024, "g2-standard-24", 2, L4Gpu),
+    NodeInstanceMapKey("g2-standard-32") ->
+      InstanceInfo(32, 128 * 1024, "g2-standard-32", 2, L4Gpu),
+    NodeInstanceMapKey("g2-standard-48") ->
+      InstanceInfo(48, 192 * 1024, "g2-standard-48", 4, L4Gpu),
+    NodeInstanceMapKey("g2-standard-96") ->
+      InstanceInfo(96, 384 * 1024, "g2-standard-96", 8, L4Gpu),
```

Comment:
QQ: we only added the `g2-standard` series now. Is this map intended to include all GPU instance types?